src/SNPN/Common/NetworkService.cs

```diff
@@ -19,21 +23,42 @@ public class NetworkService : INetworkService, IDisposable
 		private readonly AppConfiguration _config;
 		private readonly HttpClient _httpClient;
 		private readonly ILogger _logger;
+		private readonly IUserRepo _userRepo;
 
 		private readonly AsyncPolicy _retryPolicy;
 
+		private static Object firebaseAppLock = new Object();
+
 		private readonly Dictionary<NotificationType, string> _notificationTypeMapping = new Dictionary<NotificationType, string>
 		  {
 				{ NotificationType.Badge, "wns/badge" },
 				{ NotificationType.Tile, "wns/tile" },
 				{ NotificationType.Toast, "wns/toast" }
 		  };
 
-		public NetworkService(AppConfiguration configuration, ILogger logger, HttpClient httpClient)
+		public NetworkService(AppConfiguration configuration, ILogger logger, HttpClient httpClient, IUserRepo userRepo)
 		{
 			_config = configuration;
 			_logger = logger;
 			_httpClient = httpClient;
+			_userRepo = userRepo;
+
+			// Setup Firebase Default App with credentials only once.
+			lock(firebaseAppLock) {
```

Comment:
The only reason for this weird lock being here is that this failed in tests without it for Firebase whining about the default app being defined already. I'm guessing tests run in parallel so this oddity is here for it.

Reply:
It has been quite a while since I did .NET work, yeah I should have tried that first. I made this change as well as the other DI ones.